### PR TITLE
feat(udf): support inputs without type annotations

### PR DESCRIPTION
**Pull request by cpcloud**
Adds support for eliding inputs types to UDFs.

Depends on #7119.

### CODE REVIEW

File: ibis/expr/operations/udf.py [hunk 1]
```diff
@@ -298,10 +298,11 @@ def _make_node(
         func_name = name or fn.__name__
 
         for arg_name, param in inspect.signature(fn).parameters.items():
-            if (raw_dtype := annotations.get(arg_name)) is None:
-                raise exc.MissingParameterAnnotationError(fn, arg_name)
-
-            arg = rlz.ValueOf(dt.dtype(raw_dtype))
+            if (raw_dtype := annotations.get(arg_name)) is not None:
```
Comment (kszucs) [inline ibis/expr/operations/udf.py:301]: `pattern = rlz.ValueOf(annotations.get(arg_name))` should be sufficient.

Comment (cpcloud) [reply ibis/expr/operations/udf.py:301]: Ah, good call!

Comment (cpcloud) [reply ibis/expr/operations/udf.py]: Fixed!

Comment (kszucs) [reply ibis/expr/operations/udf.py]: Did you push it, I still see the previous diff.

**Review by kszucs — APPROVED** [review]
After rebasing and pushing the missing change it is good to go.

File: ibis/expr/operations/udf.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/operations/udf.py]: Oh, lemme push this up.

Comment (cpcloud) [reply ibis/expr/operations/udf.py]: Ah, it's pushed now!

**Pull request merged**